staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/queueset.go

```diff
@@ -472,16 +472,18 @@ func (qs *queueSet) timeoutOldRequestsAndRejectOrEnqueueLocked(ctx context.Conte
 // using the given hashValue and the shuffle sharding parameters of the queueSet.
 func (qs *queueSet) chooseQueueIndexLocked(hashValue uint64, descr1, descr2 interface{}) int {
 	bestQueueIdx := -1
-	bestQueueLen := int(math.MaxInt32)
+	bestQueueWidth := int(math.MaxInt32)
 	// the dealer uses the current desired number of queues, which is no larger than the number in `qs.queues`.
 	qs.dealer.Deal(hashValue, func(queueIdx int) {
-		thisLen := qs.queues[queueIdx].requests.Length()
-		klog.V(7).Infof("QS(%s): For request %#+v %#+v considering queue %d of length %d", qs.qCfg.Name, descr1, descr2, queueIdx, thisLen)
-		if thisLen < bestQueueLen {
-			bestQueueIdx, bestQueueLen = queueIdx, thisLen
+		// TODO: Consider taking into account `additional latency` of requests
+		// in addition to their widths.
```

Comment:
Yes, I think the ideal would be to judge based on the current projected E (completion time in the virtual world) of the youngest request in the queue, since that is (the best available estimate of) what the virtual world scheduling of this new request will be based on.

Reply:
Extended the comment in that PR too.